integration-tests/simnode/src/main.rs

```diff
@@ -0,0 +1,80 @@
+#![deny(unused_extern_crates, missing_docs)]
+
+//! Basic example of end to end runtime tests.
+mod chain_info;
+mod node;
+
+pub use chain_info::*;
+use picasso_runtime::Event;
+use sc_client_api::{call_executor::ExecutorProvider, CallExecutor};
+use sp_blockchain::HeaderBackend;
+use sp_runtime::generic::BlockId;
+use sp_runtime::AccountId32;
+use std::error::Error;
+use std::str::FromStr;
+
+fn main() -> Result<(), Box<dyn Error>> {
+	node::run(|node| async move {
+		let from = AccountId32::from_str("5uAfQTqudXnnSgSMPVowwRjgNFxBDW2d5AQXP2vHDHy2yJ4w")?;
+
+		node.submit_extrinsic(
+			frame_system::Call::remark { remark: b"Hello World".to_vec() },
+			Some(from),
+		)
+		.await?;
+		node.seal_blocks(1).await;
+
+		let old_runtime_version = node
+			.client()
+			.executor()
+			.runtime_version(&BlockId::Hash(node.client().info().best_hash))?
+			.spec_version;
+		println!("\n\nold_runtime_version: {}\n\n", old_runtime_version);
+
+		let code = picasso_runtime::WASM_BINARY
+			.ok_or("Polkadot development wasm not available")?
+			.to_vec();
+
+		let call = sudo::Call::sudo_unchecked_weight {
+			call: Box::new(frame_system::Call::set_code { code }.into()),
+			weight: 0,
+		};
+		// let su = AccountId32::from_str("5z93WG1Lz47b8AjtVbMaLC4M8rohecXPMSjRDBaMUAbmeCi7")?;
+		let su = AccountId32::from_str("5uAfQTqudXnnSgSMPVowwRjgNFxBDW2d5AQXP2vHDHy2yJ4w")?;
+		node.submit_extrinsic(call, Some(su)).await?;
+		node.seal_blocks(2).await;
+		// assert that the runtime has been updated by looking at events
+		let events = node
+			.events()
+			.into_iter()
+			.filter(|event| match event.event {
+				Event::ParachainSystem(parachain_system::Event::ValidationFunctionApplied(_)) => {
+					true
+				}
+				_ => false,
+			})
+			.collect::<Vec<_>>();
+		// make sure event was emitted
+		assert_eq!(
+			events.len(),
+			1,
+			"system::Event::CodeUpdate not found in events: {:#?}",
+			node.events()
+		);
+		let new_runtime_version = node
+			.client()
+			.executor()
+			.runtime_version(&BlockId::Hash(node.client().info().best_hash))?
+			.spec_version;
+		println!("\n\nnew_runtime_version: {}\n\n", new_runtime_version);
+
+		// just confirming
+		assert!(
+			new_runtime_version > old_runtime_version,
```

Comment:
and new version has version as it is defined in code?

Reply:
as defined in spec version